statsmodels/survey/survey_model.py

```diff
@@ -0,0 +1,19 @@
+import numpy as np
+import statsmodels.api as sm
+import summary_stats as ss
+
+# need to put in init_args and fit_args. Then do error checking for
+# model_class, args, and y
+
+class SurveyModel(object):
+
+    def __init__(self, design, model_class, y, init_args=None, fit_args=None):
+        self.design = design
+        self.model = model_class
+        self.init_args = init_args
+        self.fit_args = fit_args
+        self.y = y
+
+    def fit(self, data):
```

Comment:
@jarvmiller  https://github.com/statsmodels/statsmodels/issues/2848
Eventually, we should have `prob_weights` in most models, but those don't exist yet.
At the current stage it's possible to use freq_weights in GLM, and weights in WLS.
Application to other models will need to wait until they have an appropriate weights option.